### ISSUE TITLE
Error: Error response from daemon: ps: ps: invalid option -- 'e'

### ISSUE THREAD

**Issue by ab77** [balena-os/balena-engine#236]
**Steps to reproduce the issue:**
1. run `balena top abcdef123456` against a running container

**Describe the results you received:**
```
# balena top abcdef123456
Error response from daemon: ps: ps: invalid option -- 'e'
```

**Describe the results you expected:**
```
$ baena top abcdef123456
PID                 USER                TIME                COMMAND
25637               root                0:00                sh
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `balena-engine version`:**

```
# balena version
Client:
 Version:           19.03.13-dev
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        074a481789174b4b6fd2d706086e8ffceb72e924
 Built:             Sun Aug 16 19:28:03 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.13-dev
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       074a481789174b4b6fd2d706086e8ffceb72e924
  Built:            Sun Aug 16 19:28:03 2020
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.2.0+unknown
  GitCommit:        
 runc:
  Version:          
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 balena-engine-init:
  Version:          0.13.0
  GitCommit:        949e6fa-dirty
```

**Output of `balena-engine info`:**

```
# balena info   
Client:
 Debug Mode: false

Server:
 Containers: 8
  Running: 5
  Paused: 0
  Stopped: 3
 Images: 17
 Server Version: 19.03.13-dev
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: journald
 Cgroup Driver: systemd
 Plugins:
  Volume: local
  Network: bridge host null
  Log: journald json-file local
 Swarm: 
  NodeID: 
  Is Manager: false
  Node Address: 
 Runtimes: bare runc
 Default Runtime: runc
 Init Binary: balena-engine-init
 containerd version: 
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: 949e6fa-dirty (expected: fec3683b971d9)
 Kernel Version: 5.2.10-yocto-standard
 Operating System: balenaOS 2.58.6+rev1
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 62.9GiB
 Name: c678c51
 ID: K3TL:2X5P:MCPB:ZDKU:KCQA:6ABL:DJU4:YKXF:T7ZE:Y35M:ZESN:5R2D
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (device type, OS, etc.):**

N/A

**Comment by robertgzr** [comment]
The `balena top` command is implemented by shelling out to the `ps` command on the system: https://github.com/balena-os/balena-engine/blob/19.03-balena/daemon/top_unix.go#L167
This is because balenaOS is using busybox's ps, which doesn't support the '-e' flag.

It would be interesting to see if the windows behavior can be leveraged for busybox systems: https://github.com/balena-os/balena-engine/blob/19.03-balena/daemon/top_windows.go#L27

**Comment by klutchell** [comment]
Oddly, according to the busybox source for 1.30.1, the -e flag and others should be supported (and ignored). So I'm not sure why we see the error on balenaOS but not on docker.io/busybox:1.30.1.

https://git.busybox.net/busybox/tree/procps/ps.c?h=1_30_1#n577

I suspect it's related to our busybox build config, but it's not obvious which option we would be missing as this looks like core functionality. (@alexgg?)

We have a few options to approach this:
1. change meta-balena busybox defconfig to behave more like procps
2. change balena-machine top command to be more platform agnostic by trying multiple variations
3. change balena-machine top command to be more platform agnostic by supporting the lowest common flags/fields

**Comment by klutchell** [comment]
Leaning on option 3, if we assume that the local ps command supports no args at all (including specifying pids) we can try to call `ps` with no args and filter the pids we want via `ParsePSOutput`. In this case we also wouldn't make any assumptions about headers, columns, etc. and just print WYSIWYG. This is a slightly bigger change but offers the maximum compatibility.

**Comment by robertgzr** [comment]
Fix for this is in [v19.03.15](https://github.com/balena-os/balena-engine/blob/master/CHANGELOG.md#v190315)
